### PR TITLE
Upgrade Loki dependency to v2.6.1

### PR DESCRIPTION
**Pull request by rfratto**

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -430,12 +419,19 @@ func load(fs *flag.FlagSet, args []string, loader loaderFunc) (*Config, error) {
...
-	err := LoadBytes([]byte(rawCfg), false, cfg)
+	var cfg Config
+	err := LoadBytes([]byte(rawCfg), false, &cfg)
 	require.NoError(t, err)
-	bb, err := yaml.Marshal(cfg)
+
+	// Set integrations version to make sure our marshal function goes through
+	// the custom marshaling code.
+	err = cfg.Integrations.setVersion(integrationsVersion1)
 	require.NoError(t, err)
```
Comment (rfratto) [inline pkg/config/config.go:429]: I'm genuinely not sure why this worked before. The version needs to be set for remarshaling to work properly.

File: pkg/config/config.go [hunk 2]
```diff
@@ -430,12 +419,19 @@ func load(fs *flag.FlagSet, args []string, loader loaderFunc) (*Config, error) {
...
-	scrubbedCfg := string(bb)
-	require.True(t, strings.Contains(scrubbedCfg, "<secret>"))
-	require.False(t, strings.Contains(scrubbedCfg, originalValue))
+
+	bb, err := yaml.Marshal(&cfg)
+	require.NoError(t, err)
+
+	fmt.Println(string(bb))
```
Comment (rfratto) [inline pkg/config/config.go:434]:
```suggestion
```

File: pkg/config/integrations.go [hunk 3]
```diff
@@ -62,7 +62,7 @@ func (c VersionedIntegrations) MarshalYAML() (interface{}, error) {
 	case c.configV2 != nil:
 		return c.configV2, nil
 	default:
-		return c.raw, nil
+		return &c.raw, nil
```
Comment (rfratto) [inline pkg/config/integrations.go:65]:
```suggestion
		// A pointer is needed for the yaml.Marshaler implementation to work.
		return &c.raw, nil
```

**Review by mattdurham — APPROVED** [review]
LGTM

**Pull request merged**